packages/keyring-controller/src/KeyringController.ts

```diff
@@ -751,10 +751,6 @@ export class KeyringController extends BaseController<
 
     const keyring = await this.#newKeyring(type, opts);
 
-    if (!keyring) {
-      throw new Error(KeyringControllerError.NoKeyring);
-    }
```

Comment:
This condition is not possible, even in case the keyring builder is somehow broken and returns `undefined`, as `#newKeyring` would throw an error.